sys/fuchsia/fidl.txt

```diff
@@ -23,6 +23,11 @@ type fidl_message_header[METHOD_ORDINAL] {
 	ordinal		const[METHOD_ORDINAL, int32]
 }
 
+type fidl_union[T] {
```

Comment:
This will give us mismatching tag/object type, right? I think it's important to make them matching, because if the tag is wrong parsing fails, if tag does not match object everything is mis-parsed.
Is should also be relative simple to support matching tag/type. Instead of:

```
type fidl_union[T] {
	tag	int32
	union	T
}

	infoInLine	fidl_union[fuchsia_io_ObjectInfoInLine]
	
fuchsia_io_ObjectInfoInLine [
	serviceInLine	fuchsia_io_ServiceInLine
	fileInLine	fuchsia_io_FileObjectInLine
	directoryInLine	fuchsia_io_DirectoryObjectInLine
	pipeInLine	fuchsia_io_PipeInLine
	vmofileInLine	fuchsia_io_VmofileInLine
	deviceInLine	fuchsia_io_DeviceInLine
]
```

We can do something along the lines of:
```
type fidl_union[TAG, DATA] {
	tag	const[TAG, int32]
	data	DATA
}

	infoInLine	fuchsia_io_ObjectInfoInLine
	
fuchsia_io_ObjectInfoInLine [
	serviceInLine	fidl_union[0, fuchsia_io_ServiceInLine]
	fileInLine	fidl_union[1, fuchsia_io_FileObjectInLine]
...
]
```